src/core/util/ElfReader.cc

```diff
@@ -352,8 +402,56 @@ void ElfReader::printMangled() {
 
 void ElfReader::printSections() {
 	for (container_t::const_iterator it = m_sectiontable.begin(); it !=m_sectiontable.end(); ++it) {
-		m_log  << "0x"  << it->getAddress() << "\t" << it->getName().c_str() << "\t" << it->getSize() << std::endl;
+		m_log  << "0x"	<< it->getAddress() << "\t" << it->getName().c_str() << "\t" << it->getSize() << std::endl;
 	}
 }
 
+void ElfReader::printSegments() {
+	for(auto it = m_segmenttable.begin(); it != m_segmenttable.end(); ++it) {
+		m_log << *it << std::endl;
+	}
+}
+
+std::pair<guest_address_t,guest_address_t>
+ElfReader::getValidAddressBounds() {
+	guest_address_t min = std::numeric_limits<guest_address_t>::max();
+	guest_address_t max = std::numeric_limits<guest_address_t>::min();
+
+	for(const auto& seg: m_segmenttable) {
+		min = std::min(min, seg.getStart());
+		max = std::max(max, seg.getEnd());
+	}
+	assert(min != std::numeric_limits<guest_address_t>::max()
```

Comment:
Ggf. nicht nur in Debug-Builds abfangen.

Reply:
Done. Ich werfe dann mal  invalid_argument assertions.